src/04-spin-boson-model.md

```diff
@@ -0,0 +1,800 @@
+df.head()
+```
+
+We see that we expect to calculate eigenvalues for 10 levels - these are associated with the 2 states from the TSS (|+>, |->) and 5 bosons states ($n= 0,1,2,3,4$) making $2\times5 = 10$ levels.
+
+
+### 4.2.1 - Spin-boson landscape $U=0$
+
+
+We're now ready to fill in the dataframe with the energy of the stationary sates by iterating over the rows using the [`iterrows`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.iterrows.html) function.
+
+Throughout this tutorial we will set $\omega=1$.
+
+```python
+for i, row in df.iterrows():
+    H =  row["$\Delta E$"]*two_state + 1*bosons
+    evals, ekets = H.eigenstates()
+    df.iloc[i,1:] = evals
+```
+
+Because we stored the data in a dataframe, plotting is now easy.
+
+```python
+df.plot(x="$\Delta E$",figsize=(10,8),ylim=[-0.5,5.5],legend=True, 
+        title="Stationary states for $H = (\Delta E / 2) \sigma_z + \hbar\omega(a^{{\dagger}}a +1/2)$     ($\omega=1$)     (Fig 1)");
+plt.ylabel("Energy");
+```
+
+How do we understand Fig 1 (often called an energy level diagram)?
+
+Start by focusing on where $\Delta E = 0$, i.e. there is no difference between the energies of the |+> and |-> states - this is the first thing we looked at back in Tutorial 1. There are several states/levels that appear to cross each other, they correspond to:
+- orange/blue - 0 bosons (|0,±>)
+- red/green - 1 boson (|1,±>)
+- brown/purple - 2 bosons (|2,±>)
+- grey/pink - 3 bosons (|3,±>)
+- blue/yellow - 4 bosons (|4,±>)
+
+Let's take the orange (|0,+>) and green (|1,->) lines. As we increase $\Delta E$, The energy of |0,+> goes up  and |1,-> goes down. Eventually, these levels end up with the same energy (they cross) despite having a different number of bosons - this is what allows the |0,+> to couple strongly to |1,-> i.e. it's what makes possible the spontaneous emission we saw in the last tutorial. 
+
+What's particularly interesting is the location of the crossings. For example, the crossing we just described happens when $\Delta E = \omega = 1$. This is physically significant because it is identical to the resonance condition we discovered in [Tutorial 2](https://nbviewer.jupyter.org/github/project-ida/two-state-quantum-systems/blob/master/02-perturbing-a-two-state-system.ipynb#Resonance) when the driving frequency of the classical field matched the transition frequency of the TSS. 
+
+In Tutorial 2, we found this resonance somewhat by accident. Now though, by looking at the other crossings in Fig 1 we can already start to predict the existence of other resonances when $\Delta E = 2\omega, 3\omega $ etc (we might call these "non-primary" resonances). We don't yet know what will happen at those resonances (maybe nothing at all 🤷) but it is not unreasonable to expect that when we switch the interaction on ($U\neq 0$) there will be some kind of interaction and we'll see the formation of anti-crossings similar to Tutorial 2.
```

Comment:
Can't we already see that there are anti-crossings and expected state changes?

Reply:
There are no anti-crossings when there is no coupling between the boson field and TSS. It's only when we turn interaction on (U non zero) that we can hope to see anti-crossigs. At this point we just see a lot of crossings which motiviates us to think that maybe they'll all turn into anti-crossings when we turn interaciton on

Reply:
I guess I'm confused because of the coloring of the lines. For instance when tracing the orange line, we can see it almost runs into the green one and then bends downward again, moving away from it. Isn't this anti-crossing? I would consider it a crossing if the orange line went straight up and through (like in Nicola's drawing). Let me know if I don't have the basic terms down properly here.

Reply:
I understand why there is confusion based on the colours. We'll need to discuss that a bit more I think. 

Basically, if they touch then it's not an anti-crossing, it is a crossing.